### PR TITLE
GET Available Connectors [#706]

### LINKED ISSUE #752: Retrieve list of available connectors (Backend)
### Is your feature request related to a specific problem?

As a Fidesops admin user, need the ability to see a list of available connector types to choose from so that I can select one to connect my fidesops environment to.

### Additional Context
- this is a backend ticket only to support an eventual frontend
- backend must support searching for a connector by connector name (i.e. "posgresql" or "hubpost")
- must return all types of connectors regardless of database vs saas type
- should support partial searches (i.e. if user entered "hub", then the hubspot and github adaptor should return (assuming we have both of those connectors available))

### PR DESCRIPTION
**Pull request by pattisdr**
❗ Dependent on #706 - Merge that first

# Purpose
As a Fidesops admin user, need the ability to see a list of available connector types to choose from so that I can select one to connect my fidesops environment to.

# Changes
-  Adds an endpoint to get available connectors that are ready to go (this is a combination of datastore and saas connectors minus ("manual", "custom", generic "saas", and "https" connectors.) `GET api/v1/connection_type`
    - Can be filtered on whether search param is a substring of the connector name `GET api/v1/connection_type?search=red`

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [x] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #752

### CODE REVIEW

File: src/fidesops/api/v1/endpoints/connection_type_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    def is_match(elem: str) -> bool:
+        """If a search query param was included, is it a substring of an available connector type?"""
+        return search in elem if search else True
+
+    database_types: List[str] = [
+        conn_type.value
+        for conn_type in ConnectionType
+        if conn_type
+        not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
```
Comment (pattisdr) [inline src/fidesops/api/v1/endpoints/connection_type_endpoints.py:38]: my assumption is that these types and the "custom" type below shouldn't be included in this list, that these are connectors that are available and ready to go

File: src/fidesops/api/v1/endpoints/connection_type_endpoints.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from fidesops.util.oauth_util import verify_oauth_client
+
+router = APIRouter(tags=["Connection Types"], prefix=V1_URL_PREFIX)
+
+logger = logging.getLogger(__name__)
+
+
+@router.get(
+    CONNECTION_TYPES,
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_TYPE_READ])],
```
Comment (pattisdr) [inline src/fidesops/api/v1/endpoints/connection_type_endpoints.py:22]: I set scopes here but this may not be necessary

**Review by seanpreston — APPROVED** [review]

**Comment by pattisdr** [comment]
@ethyca/docs-authors Small docs addition added to the guides about how to see a list of available connector types

**Review by conceptualshark — APPROVED** [review]
👍 good on the docs side of the house!

**Comment by pattisdr** [comment]
fixing ClientDetail model import..

**Pull request merged**